geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala

```diff
@@ -684,6 +684,31 @@ class AccumuloDataStore(val connector: Connector,
     GeoMesaTable.getTableNames(sft, this).filter(tableOps.exists).foreach(tableOps.delete)
 
   /**
+    * Checks that the distributed runtime jar matches the project version of this client. We cache
+    * successful checks for 10 minutes.
+    */
+  private def checkProjectVersion(): Unit = {
+    if (projectVersionCheck.get() < System.currentTimeMillis()) {
+      val clientVersion = GeoMesaProperties.GeoMesaProjectVersion
+      val scanner = connector.createScanner(catalogTable, new Authorizations())
+      val iteratorVersion = try {
+        ProjectVersionIterator.scanProjectVersion(scanner)
+      } catch {
+        case NonFatal(e) => "unavailable"
+      } finally {
+        scanner.close()
+      }
+      if (iteratorVersion != clientVersion) {
```

Comment:
Yeah, it would prevent older clients from writing - that may be a good thing, as if they tried to do an update it might fail. If they really want to keep using the old clients, they can always use the ignore flag.